pip/rules.bzl

```diff
@@ -145,25 +153,35 @@ def _assemble_pip_impl(ctx):
     args.add_all("--imports", imports)
 
     ctx.actions.run(
-        inputs = [version_file, setup_py, ctx.file.long_description_file, ctx.file.requirements_file] + python_source_files,
-        outputs = [ctx.outputs.pip_package],
+        inputs = [version_file, setup_py, ctx.file.long_description_file, ctx.file.requirements_file] + python_source_files + data_files,
+        outputs = [ctx.outputs.pip_package, ctx.outputs.pip_wheel],
         arguments = [args],
         executable = ctx.executable._assemble_script,
     )
 
-    return [PyDeploymentInfo(package=ctx.outputs.pip_package, version_file=version_file)]
+    return [PyDeploymentInfo(package=ctx.outputs.pip_package, wheel=ctx.outputs.pip_wheel, version_file=version_file)]
 
 
 def _deploy_pip_impl(ctx):
     deployment_script = ctx.actions.declare_file("{}_deploy.py".format(ctx.attr.name))
+    if ctx.attr.snapshot and ctx.attr.release:
+        fail(msg="canot deploy snapshot and release at the same time.")
```

Comment:
i wonder if you'll get push-back when you try to upstream this check as it's a change from the previous behavior

Reply:
If you read their implementation, this is actually the expected behavior. The original rule will run into an error due to the arg of "repo_type_key"  is missing. They cannot run twine command with both urls.

Reply:
awesome ... ❤️  it